build/VSL.Imports.Closed.targets

```diff
@@ -1,4 +1,35 @@
 <Project DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  
+  <!-- ====================================================================================
+  
+         Support for in-place modification of the compiled binary.
+         Since there may be more than one post compile modification, we must take care not
+         to break incremental builds. A timestemp file is written out when all modification
+         targets have completed.
+         Note that the targets participating in post compile modification must list the
+         sentinel file as one of their outputs, but they should not modify it.
+
+       ==================================================================================== -->
+
+  <PropertyGroup>
+    <PostCompileBinaryModificationSentinelFile>$(IntermediateOutputPath)$(TargetFileName).pcbm</PostCompileBinaryModificationSentinelFile>
+    <CompileDependsOn>$(CompileDependsOn);PostCompileBinaryModification</CompileDependsOn>
+    <CleanDependsOn>$(CleanDependsOn);CleanPostCompileBinaryModification</CleanDependsOn>
+  </PropertyGroup>
+
+  <Target Name="PostCompileBinaryModification"
+          DependsOnTargets="ApplyOptimizations;FakeSign"
+          Inputs="@(IntermediateAssembly)"
+          Outputs="@(IntermediateAssembly);$(PostCompileBinaryModificationSentinelFile)">
+
+    <!-- Write out a sentinel timestamp file to prevent unnecessary work in incremental builds. -->
+    <Touch AlwaysCreate="true" Files="$(PostCompileBinaryModificationSentinelFile)" />
+  </Target>
+
+  <Target Name="CleanPostCompileBinaryModification">
+    <Delete Files="$(PostCompileBinaryModificationSentinelFile)" />
```

Comment:
The easier option is to put the ...SentinelFile item into the "FileWrites" item group. Those get stored and clean will delete it automatically.